package/MDAnalysis/analysis/dihedrals.py

```diff
@@ -0,0 +1,121 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import warnings
+
+from MDAnalysis.analysis.base import AnalysisBase
+
+def dihedral_calc(atomgroups):
+    """Calculates phi and psi angles for a list of AtomGroups over trajectory.
+
+    Parameters
+    ----------
+    atomgroups : list of AtomGroups
+        must be a list of one or more AtomGroups containing 5 atoms in the
+        correct order (i.e. C-N-CA-C-N)
+
+    Returns
+    -------
+    angles : numpy.ndarray
+        An array of time steps which contain (phi,psi) for all AtomGroups.
+    """
+
+    dihedrals = [atomgroup.dihedral for atomgroup in atomgroups]
+    angles = [dih.value() for dih in dihedrals]
```

Comment:
This line isn't wrong, but it would be better to use `lib.distances.calc_dihedrals` and do them all at once

Reply:
Oh, that works?

Great!

Reply:
@hfmull see https://www.mdanalysis.org/mdanalysis/documentation_pages/lib/distances.html#MDAnalysis.lib.distances.calc_dihedrals

Reply:
@richardjgowers does `dih.value()` take PBC into account by default?

[lib.distances.calc_dihedrals](https://www.mdanalysis.org/mdanalysis/documentation_pages/lib/distances.html#MDAnalysis.lib.distances.calc_dihedrals) will want a `box` from [AtomGroup.dimensions](https://www.mdanalysis.org/mdanalysis/documentation_pages/core/groups.html#MDAnalysis.core.groups.AtomGroup.dimensions).

Reply:
@hfmull to use `calc_dihedrals` you will need "vertical atomgroups", ie 4 atomgroups, one representing the first atom in each dihedral, the next the second atom in each dihedral.  I'd make these in the `__init__` of your class.  Then you can call `calc_dihedrals(ag1.positions, ag2.positions, ag3.positions, ag4.positions, box=ag1.dimensions)`